BattleBump/HomeViewController.swift

```diff
@@ -9,283 +9,265 @@
+            self.performSegue(withIdentifier: "edit", sender: self)
+        } else {
+            return
+        }
+    }
     
-    guard let text = sender.text, !text.isEmpty else {
-      return
+    @IBAction func hostButtonPressed(_ sender: UIButton) {
+        mpcManager.advertiseToPeers()
     }
     
-    if ((sender.text?.containsOnlyEmoji)! && (sender.text?.isSingleEmoji)!) {
-      let defaults = UserDefaults.standard
-      defaults.set(sender.text, forKey: "playerEmoji")
-    } else {
-      let alert = UIAlertController(title: "One emoji at a time 💩", message: nil, preferredStyle: .alert)
-      let defaultAction = UIAlertAction(title: "OK", style: .default, handler: {(alert: UIAlertAction!) in
-        sender.text = "🙃"
-        self.playerEmojiTextField.resignFirstResponder()
-      })
-      
-      alert.addAction(defaultAction)
-      present(alert, animated: true, completion: nil)
+    @IBAction func playerNameTextFieldDidEndEditing(_ sender: UITextField) {
+        
+        guard let text = sender.text, !text.isEmpty, playerName != sender.text else {
+            return
+        }
+        
+        playerName = text
+        me.name = text
+        let defaults = UserDefaults.standard
+        defaults.set(text, forKey: "playerName")
+        playerNameTextField.resignFirstResponder()
+        
     }
     
-  }
-  
-  //MARK: - Functions -
-  
-  func loadUserDefaults() {
-    
-    let defaults = UserDefaults.standard
-    if(defaults.dictionaryRepresentation().keys.contains("playerName")) {
-      playerName = defaults.string(forKey: "playerName")
-      playerEmoji = defaults.string(forKey:"playerEmoji")
-    } else {
-      playerName = "Player"
-      playerEmoji = "😎"
+    //MARK: - Functions -
+    
+    func loadUserDefaults() {
+        
+        let defaults = UserDefaults.standard
+        if(defaults.dictionaryRepresentation().keys.contains("playerName")) {
+            playerName = defaults.string(forKey: "playerName")
+        } else {
+            playerName = "Player"
+        }
+        
+        playerNameTextField.text = playerName
+        
+        if defaults.dictionaryRepresentation().keys.contains("One") {
+            movesetOne = defaults.object(forKey: "One") as? Moveset
+            movesetTwo = defaults.object(forKey: "Two") as? Moveset
+            movesetThree = defaults.object(forKey: "Three") as? Moveset
+        } else {
+            movesetOne = Moveset(name: "Standard", numberOfMoves: 3, movesArray: ["Rock", "Paper", "Scissors"])
+            movesetTwo = Moveset(name: "Weapon triangle", numberOfMoves: 3, movesArray: ["Sword", "Spear", "Axe"])
+            movesetThree = Moveset(name: "Pokemon", numberOfMoves: 7, movesArray: ["Grass", "Fire", "Rock", "Psychic", "Fighting", "Flying", "Water"])
+        }
+        
+        movesetArray = [movesetOne!, movesetTwo!, movesetThree!]
+        movesetButtonArray = [movesetOneButton, movesetTwoButton, movesetThreeButton]
```

Comment:
To remove `movesetOne/Two/Three`:
```suggestion
        if let movesetOne = defaults.object(forKey: "One") as? Moveset, let movesetTwo = defaults.object(forKey: "Two") as? Moveset, let movesetThree = defaults.object(forKey: "Three") as? Moveset {
            movesetArray = [movesetOne, movesetTwo, movesetThree]
        } else {
            movesetArray = [
                Moveset(name: "Standard", numberOfMoves: 3, movesArray: ["Rock", "Paper", "Scissors"]),
                Moveset(name: "Weapon triangle", numberOfMoves: 3, movesArray: ["Sword", "Spear", "Axe"]),
                Moveset(name: "Pokemon", numberOfMoves: 7, movesArray: ["Grass", "Fire", "Rock", "Psychic", "Fighting", "Flying", "Water"]),
            ]
        }
        
        movesetButtonArray = [movesetOneButton, movesetTwoButton, movesetThreeButton]
```